default.nix

```diff
@@ -0,0 +1,84 @@
+# Building purebred
+#
+# You should be able to simply invoke:
+#
+# $ nix-build
+#
+# or, to be explicit:
+#
+# $ nix-build default.nix
+#
+# in purebred's root directory in order to build purebred. You'll find the binary under:
+#
+# $ ls result/bin/purebred
+#
+# if the build was successful.
+#
+#
+# Choosing a different compiler than the default
+#
+# In order to choose a different compiler, invoke nix build like so (escaping
+# the quotes is needed, since we're passing a string literal):
+#
+# $ nix-build --arg compiler \"ghc442\"
+#
+#
+# Use as a development environment
+#
+# $ nix-shell default.nix
+#
+{ compiler ? "ghc822" }:
+
+let
+  config = {
+    packageOverrides = super: let self = super.pkgs; in
+    {
+      haskell = super.haskell // {
+        packages = super.haskell.packages // {
+          "${compiler}" = super.haskell.packages."${compiler}".override {
+            overrides = self: super: {
+              # Build with latest known stable version.
+              purebred = with pkgs.haskell.lib; dontHaddock (self.callPackage ./.nix-helpers/purebred.nix { });
```

Comment:
Why not build the haddock?  Is there a technical reason why we disable it?

Reply:
No technical reason. Just took it out to save a few seconds built time. Will remove it.

Reply:
Done.